xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "RPProcessInfo.h"
+#include "cores/DataCacheCore.h"
+
+extern "C" {
+#include "libavutil/pixdesc.h"
+}
+
+using namespace KODI;
+using namespace RETRO;
+
+CRPProcessInfo* CRPProcessInfo::CreateInstance()
+{
+  return new CRPProcessInfo();
+}
+
+void CRPProcessInfo::SetDataCache(CDataCacheCore *cache)
+{
+  m_dataCache = cache;;
+}
+
+void CRPProcessInfo::ResetInfo()
+{
+  if (m_dataCache != nullptr)
+  {
+    m_dataCache->SetVideoDecoderName("", false);
+    m_dataCache->SetVideoDeintMethod("");
+    m_dataCache->SetVideoPixelFormat("");
+    m_dataCache->SetVideoDimensions(0, 0);
+    m_dataCache->SetVideoFps(0.0f);
+    m_dataCache->SetVideoDAR(1.0f);
+    m_dataCache->SetAudioDecoderName("");
+    m_dataCache->SetAudioChannels("");
+    m_dataCache->SetAudioSampleRate(0);
+    m_dataCache->SetAudioBitsPerSample(0);
+    m_dataCache->SetRenderClockSync(false);
+    m_dataCache->SetStateSeeking(false);
+    m_dataCache->SetSpeed(1.0f, 1.0f);
+    m_dataCache->SetGuiRender(true); //! @todo
```

Comment:
@FernetMenta What is gui render? This isn't documented.

Reply:
Presumably it's for when rendering the GUI pass in VP/RM (`Render` is called with the `gui` parameter set to true).
Likewise, `SetVideoRender` below is probably for when rendering the output frame. But I don't know if you just need to set both to true once, or update them multiple times every frame while rendering.
As with anything from `CDataCacheCore`, updating it is optional.